plugins/org.python.pydev/tests_completions/org/python/pydev/ast/codecompletion/PythonCompletionWithBuiltinsPython3Test.java

```diff
@@ -239,4 +239,23 @@ public void testCodeCompletionPep484Return() throws Exception {
         requestCompl(s, s.length(), -1, new String[] { "title()", "translate(table)" });
     }
 
+    public void testCodeCompletionPep484Return2() throws Exception {
+        String s;
+        s = ""
+                + "from typing import List\n" +
+                "\n" +
+                "class Bar(object):\n" +
+                "    def bar(self):\n" +
+                "        pass\n" +
+                "\n" +
+                "def list_bar() -> List[Bar]:\n" +
```

Comment:
Actually, re-reading the typing info here should not be provided in the method return (it's already given later on as `a:Bar`).

So, this should really be:

```
class Bar(object):

    def bar(self):
        pass

def list_bar():
    pass

def something():
    lst = list_bar()
    a:Bar = lst[0]
    a.
```